docs/content.zh/docs/dev/datastream/fault-tolerance/state_v2.md

```diff
@@ -0,0 +1,601 @@
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Working with State V2 (New APIs)
+
+In this section you will learn about the new APIs that Flink provides for writing
+stateful programs. Please take a look at [Stateful Stream
+Processing]({{< ref "docs/concepts/stateful-stream-processing" >}})
+to learn about the concepts behind stateful stream processing.
+
+The new state API is designed to be more flexible than the previous API. User can perform
+asynchronous state operations, thus making it more powerful and more efficient.
+The asynchronous state access is essential for the state backend to be able to handle
+large state sizes and to be able to spill to remote file systems when necessary. 
+
+## Keyed DataStream
+
+If you want to use keyed state, you first need to specify a key on a
+`DataStream` that should be used to partition the state (and also the records
+in the stream themselves). You can specify a key using `keyBy(KeySelector)`
+in Java API on a `DataStream`. This will yield a `KeyedStream`, which then allows operations
+that use keyed state. You should perform `enableAsyncState()` on the `KeyedStream` to enable
+asynchronous state operations.
+
+A key selector function takes a single record as input and returns the key for
+that record. The key can be of any type and **must** be derived from
+deterministic computations.
+
+The data model of Flink is not based on key-value pairs. Therefore, you do not
+need to physically pack the data set types into keys and values. Keys are
+"virtual": they are defined as functions over the actual data to guide the
+grouping operator.
+
+The following example shows a key selector function that simply returns the
+field of an object:
+
+{{< tabs "54e3bde7-659a-4683-81fa-312a2c81036b" >}}
+{{< tab "Java" >}}
+```java
+// some ordinary POJO
+public class WC {
+  public String word;
+  public int count;
+
+  public String getWord() { return word; }
+}
+DataStream<WC> words = // [...]
+KeyedStream<WC> keyed = words
+  .keyBy(WC::getWord).enableAsyncState();
+```
+{{< /tabs >}}
+
+{{< top >}}
+
+## Using Keyed State V2
+
+Unlike the previous state API, the new state API is designed for asynchronous state access.
+Each type of state gives two versions of the API: synchronous and asynchronous. The synchronous
+API is a blocking API that waits for the state access to complete. The asynchronous API is
+non-blocking and returns a `StateFuture` that will be completed when the state access
+is done. After that a callback or following logic will be invoked (if any).
+The asynchronous API is more efficient and should be used whenever possible.
+It is highly not recommended to mixed synchronous and asynchronous state access in the same
```

Comment:
nit: It is highly not recommended -> It is highly **not** recommended

emphasize not?